tests/Agent/IntegrationTests/ContainerApplications/AwsSdkTestApp/AwsSdkExerciser/AwsSdkExerciser.cs

```diff
@@ -117,20 +118,30 @@ public async Task<IEnumerable<Message>> SQS_ReceiveMessageAsync(int maxMessagesT
             if (messageAttributeNames.Count != 1)
                 throw new Exception("Expected messageAttributeNames to have a single element");
 
-            foreach (var message in response.Messages)
+            try
             {
-                Console.WriteLine($"Message: {message.Body}");
-                foreach (var attr in message.MessageAttributes)
+                foreach (var message in response.Messages)
                 {
-                    Console.WriteLine($"MessageAttributes: {attr.Key} = {{ DataType = {attr.Value.DataType}, StringValue = {attr.Value.StringValue}}}");
+                    Console.WriteLine($"Message: {message.Body}");
+                    if (message.MessageAttributes != null)
+                    {
+                        foreach (var attr in message.MessageAttributes)
+                        {
+                            Console.WriteLine($"MessageAttributes: {attr.Key} = {{ DataType = {attr.Value.DataType}, StringValue = {attr.Value.StringValue}}}");
+                        }
+                    }
+
+                    // delete message
+                    await _amazonSqsClient.DeleteMessageAsync(new DeleteMessageRequest
+                    {
+                        QueueUrl = _sqsQueueUrl,
+                        ReceiptHandle = message.ReceiptHandle
+                    });
                 }
-
-                // delete message
-                await _amazonSqsClient.DeleteMessageAsync(new DeleteMessageRequest
-                {
-                    QueueUrl = _sqsQueueUrl,
-                    ReceiptHandle = message.ReceiptHandle
-                });
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Caught exception processing response messages: {ex.Message}");
```

Comment:
Not sure this try/catch is useful. But in any case, it can't write to the console... ;-)

Reply:
That was left over from previous debugging attempts.  However, when I removed the try/catch, I had to introduce another null check (for `response.Messages`) and return an empty list of messages if it is null, when running the tests in the "don't initialize empty collections" mode.